src/Microsoft.AspNet.Http/Authentication/DefaultAuthenticationManager.cs

```diff
@@ -29,12 +28,28 @@ void IFeatureCache.CheckFeaturesRevision()
         {
             if (_cachedFeaturesRevision != _features.Revision)
             {
-                _authentication = null;
-                _response = null;
-                _cachedFeaturesRevision = _features.Revision;
+                ResetFeatures();
             }
         }
 
+        void IFeatureCache.SetFeaturesRevision()
```

Comment:
Don't want someone to call it; as it says "mark cache as up to date; but don't check it"